pulumi/infra/observability_env_vars.py

```diff
@@ -57,13 +59,14 @@ def otel_config(
     config:
       scrape_configs:
         - job_name: 'nomad-server'
-          scrape_interval: 10s
-          scrape_timeout: 20s
-          metrics_path: '/v1/metrics?format=prometheus'
+          scrape_interval: 20s
+          scrape_timeout: 10s
+          metrics_path: '/v1/metrics'
           params:
             format: ['prometheus']
           static_configs:
-            - targets: ['localhost:4646']
```

Comment:
This doesn't work from inside a container.

Reply:
I don't want to derail this PR, or the work you've already put into solving this problem (after all, a solution that works is much better than one that doesn't!), but are you familiar with any way of configuring the containers or their networking such that we don't have to jump through so many hoops to get this wired together?

If the answer is "no, this is how we do this", that's fine... it just seems like a lot of fiddly bits to deal with.